### PR TITLE
Feature/wf 664 design review

### PR DESCRIPTION
**Pull request by claudio147**
## Pull request
This PR fixes a few design requests (described in the ticket). For the file clean-up (build-process) I used a plugin (https://www.npmjs.com/package/webpack-clean). I didn't found a solution how I can prevent webpack from creating those files - so I decide to remove them after the build process.
 
### Ticket
https://winterhalter-fenner.atlassian.net/browse/WF-659
https://winterhalter-fenner.atlassian.net/browse/WF-664
 
### Browser testing
- http://localhost:8080/styleguide/product-detail
- http://localhost:8080/styleguide/forms
- http://localhost:6060/styleguide#einfolabel
 
### Checklist
- [x] Added JSDoc and styleguide demo
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [x] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [x] Did run automated tests and linters
- [x] Did review code and documentation
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Re-assign ticket to developer

### CODE REVIEW

File: app/components/e-info-label.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
 ### Promotion
 
-```
-<e-info-label :price-type="1"/>
+```vue
+<e-info-label :price-type=1 />
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: app/components/e-info-label.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 ```
 
 ### Promotion with priceTypeEndDate
 
-```
-<e-info-label :price-type="1" price-type-end-date="1.1.1970"/>
+```vue
+<e-info-label :price-type=1 price-type-end-date="1.1.1970"/>
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: app/components/e-info-label.md [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
 ```
 
 ### New
 
-```
-<e-info-label :price-type="2"/>
+```vue
+<e-info-label :price-type=2 />
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: app/components/e-info-label.md [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
 ```
 
 ### Discount
 
-```
-<e-info-label :price-type="3" price-type-end-date="1.1.1970"/>
+```vue
+<e-info-label :price-type=3 price-type-end-date="1.1.1970"/>
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: app/components/e-info-label.md [hunk 5]
```diff
@@ -1,35 +1,35 @@
...
 ```
 
 ### While stock lasts
 
-```
-<e-info-label :price-type="4" price-type-end-date="1.1.1970"/>
+```vue
+<e-info-label :price-type=4 price-type-end-date="1.1.1970"/>
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: app/components/e-info-label.md [hunk 6]
```diff
@@ -1,35 +1,35 @@
...
 ```
 
 ### Currency discount
 
-```
-<e-info-label :price-type="5" price-type-end-date="1.1.1970"/>
+```vue
+<e-info-label :price-type=5 price-type-end-date="1.1.1970"/>
```
Comment (patric-eberle) [inline app/components/e-info-label.md]: - [ ] This is not valid in Vue/HTML. If you want to make sure `1` is a string and not a number you have to write quotes inside quotes: `:price-type="'1'"`

File: webpack.config.js [hunk 7]
```diff
@@ -125,7 +126,16 @@ module.exports = function(env = {}, options = {}) {
       new ExtractTextPlugin({
         filename: assetsSubDirectory + `css/${prefix}[name].css?[chunkhash]`, // NOTE: postcss-pipeline currently does not support query hash (https://github.com/mistakster/postcss-pipeline-webpack-plugin/issues/30)
         allChunks: true,
-      })
+      }),
+
+      // Cleans dist directory and removes specific unnecessary files
+      new WebpackCleanPlugin([
```
Comment (patric-eberle) [inline webpack.config.js]: - [ ] This is only needed in production. Please move inside `isProduction` condition (and `!hasWatcher` even better).

Comment (patric-eberle) [inline webpack.config.js]: - [ ] Please base theme array on `themes`:

```
new WebpackCleanPlugin(
  Object.keys(themes).map(theme => `${theme}.js`),
  { basePath: path.join(__dirname, 'dist/static/js') }),
);
```

Even better would be to move the setup to the beginning of the config where the variables are defined:

```
const clean = [
  ...Object.keys(themes).map(theme => `${theme}.js`),
  'any/other/files.js',
]
```

**Comment by claudio147** [comment]
@patric-eberle fixed PR issues except:

> Stylelint error: 319:5 ✖ "no-descending-specificity" has already been disabled CssSyntaxError
- Can't see this error

> According to design max-width for content should be 1380px (is now 1420px) 
- I did this because the breakpoint is 1440px and the spacing left/right is 10px

> In component styleguides we still have console errors about invalid props.
- Can't see this errors

> e-info-label padding-left/-right too big on iOS
- Can't see the problem - looks like on the design

> Product detail fails on IE11 from 768px to 1024px
- Can't see the failing (tested IE11 - win7) product-detail-page

**Comment by patric-eberle** [comment]
@claudio147 About the errors in styleguide: you need to enable "Preserve log" because vue-styleguidist clears the log on startup...

Rest is ok.

**Comment by claudio147** [comment]
@patric-eberle Prop errors in the console are fixed.

**Pull request merged**